clients/cloud/rust/README.md

```diff
@@ -0,0 +1,61 @@
+# Overview
+
+Produce messages to produce and consume messages from [Confluent Cloud](https://www.confluent.io/confluent-cloud/) using the [rust-rdkafka client for Apache Kafka](https://github.com/fede1024/rust-rdkafka).
+
+# Prerequisites
+* [Rust client for Apache Kafka](https://github.com/fede1024/rust-rdkafka#installation) installed on your machine
+* Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/) cluster
+* Local file with configuration parameters to connect to your Confluent Cloud instance ([how do I find those?](https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients)). Format the file as follows:
+
+```bash
+$ cat ~/.ccloud/example.config
+bootstrap.servers=<broker-1,broker-2,broker-3>
+sasl.username=<api-key-id>
+sasl.password=<secret-access-key>
+```
+
+# Example 1: Hello World!
+In this example, the producer writes Kafka data to a topic in Confluent Cloud. 
+Each record has a key representing a username (e.g. `alice`) and a value.
+The consumer reads the same topic from Confluent Cloud.
+
+1. Run the producer, passing in arguments for (a) the local file with configuration parameters to connect to your Confluent Cloud instance and (b) the topic name:
+
+```bash
+$ RUSTFLAGS='-C link-args=-lzstd' cargo build
```

Comment:
Why are these RUSTFLAGS needed? librdkafka should already be linked with -lzstd

Reply:
I would love to figure out how to do a build without specifying this but I spend too much time trying to get it to work and this was the workaround. Here is the error I get:

```
  = note: Undefined symbols for architecture x86_64:
            "_ZSTD_createCStream", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_initCStream", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_endStream", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_decompress", referenced from:
                _rd_kafka_zstd_decompress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_compressBound", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_isError", referenced from:
                _rd_kafka_zstd_decompress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_compressStream", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_freeCStream", referenced from:
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_getFrameContentSize", referenced from:
                _rd_kafka_zstd_decompress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_getErrorCode", referenced from:
                _rd_kafka_zstd_decompress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
            "_ZSTD_getErrorName", referenced from:
                _rd_kafka_zstd_decompress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
                _rd_kafka_zstd_compress in librdkafka_sys-347d93b86e02028e.rlib(rdkafka_zstd.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
This looks like a bug, it should be possible to automate this with a `build.rs` file
https://github.com/fede1024/rust-rdkafka/issues/125

Reply:
@gardnervickers good find! let me try that 💯